### PR TITLE
Setting up the truffle docker + e2e test

### PR DESCRIPTION
**Pull request by josojo**
In order to test this pr, pull in the git.module from the PR in dex-contracts.

and then run docker-compose up

### CODE REVIEW

**Comment by josojo** [comment]
Next steps:
Cleaning
Integrate into travis tests

**Comment by bh2smith** [comment]
This is beginning to look really great!

**Comment by bh2smith** [comment]
Would we say it is almost ready for merge (i.e. should I checkout this branch and run it?)

**Comment by josojo** [comment]
No @bh2smith, tests are still not passing, even if you see the green hock.

@fleupold 
I found the truffle docker very useful. It was nice to have the migrations automatically done. This allows straight forward running the own scripts against it.

File: event_listener/dfusion_db/snapp_event_receiver.py [hunk 1]
```diff
@@ -35,12 +35,12 @@ def save(self, event: Dict[str, Any], block_info: Dict[str, Any]) -> None:
...
-            logging.info("Successfully updated state and balances")
-        except AssertionError as exc:
-            logging.critical(
-                "Failed to record StateTransition [{}] - {}".format(exc, transition))
+        # try:
+        self.__update_accounts(transition)
+        logging.info("Successfully updated state and balances")
+        # except AssertionError as exc:
+        #    logging.critical(
+        #        "Failed to record StateTransition [{}] - {}".format(exc, transition))
```
Comment (bh2smith) [inline event_listener/dfusion_db/snapp_event_receiver.py]: Is this commented stuff still necessary?

File: event_listener/dfusion_db/snapp_event_receiver.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def save(self, event: Dict[str, Any], block_info: Dict[str, Any]) -> None:
 
         try:
             self.initialize_accounts(
-                event['maxTokens'], event['numAccounts'], state_hash)
+                event['maxTokens'], event['maxAccounts'], state_hash)
```
Comment (bh2smith) [inline event_listener/dfusion_db/snapp_event_receiver.py]: Interesting catch.

File: docker-compose.yml [hunk 3]
```diff
@@ -14,10 +14,10 @@ services:
     image: 'trufflesuite/ganache-cli:latest'
     command: ["-d"]
 #    logging:
-#      driver: "none"
+#      driver: "none"
```
Comment (bh2smith) [inline docker-compose.yml]:
```suggestion
#      driver: "none"
```

**Comment by fleupold** [comment]
After 74 commits, this should finally be ready for review @josojo, @bh2smith

**Comment by josojo** [comment]
Sauber Felix! Thank you sooo much for taking this task from me. I know it was a tough one and no Disney land journey! 

Code is very clean and also runs on my machine!

**Comment by josojo** [comment]
it's approved from my side

**Pull request merged**